shell-ui/src/navbar/Navbar.spec.js

```diff
@@ -335,5 +335,82 @@ describe('navbar', () => {
     expect(screen.queryByText(/Test/i)).toBeInTheDocument();
   });
 
+  it('should display navbar in the expected order', async () => {
+    //S
+
+    mockOIDCProvider();
+
+    render(
+      <solutions-navbar
+        oidc-provider-url="https://mocked.ingress/oidc"
+        client-id="metalk8s-ui"
+        response-type="id_token"
+        redirect-url="http://localhost:8082"
+        scopes="openid profile email groups offline_access audience:server:client_id:oidc-auth-client"
+        options={JSON.stringify({
+          main: {
+            'http://localhost:8082/': { en: 'Third', fr: 'Plateforme', order: 3 },
+            'http://localhost:8082/test': {
+              en: 'First',
+              fr: 'Test',
+              groups: ['group1', 'group2'],
+              order: 1
+            },
+            'http://localhost:8082/test3': {
+              en: 'Second',
+              fr: 'Test3',
+              groups: ['group1', 'group2'],
+              order: 2
+            },
+          },
+          subLogin: {},
+        })}
+      />,
+    );
+    //E
+    await waitForLoadingToFinish();
+    //V
+    const entries = [...screen.queryByText(/First/i).parentElement.parentElement.children].map(e => e.querySelector('a').innerHTML)
+    expect(entries).toStrictEqual(['First', 'Second', 'Third'])
+  });
+
+  it('should enqueue unordered entries', async () => {
```

Comment:
How can this work with:

```javascript
entries.sort(([_, entryA], [__, entryB]) => (entryA.order - entryB.order));
```

??? :thinking: Javascript is really a mistery to me...

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries

> The order of the array returned by Object.entries() does not depend on how an object is defined. If there is a need for certain ordering, then the array should be sorted first, like Object.entries(obj).sort((a, b) => b[0].localeCompare(a[0]));.

Reply:
Oh, that was not my question actually @ChengYanJin (but it confirms that we should **really** not rely on default ordering).

My point was that we are sorting `entries` based on `entryA.order - entryB.order`... But we are able to handle entries for which `entryC.order === undefined`, and somehow they end up at the end (which this test is verifying). This part is really mystical to me.

Reply:
OK, tested it a bit (see https://jsfiddle.net/Loxm9ck4/2/), it's actually broken, the `sort` logic just doesn't move elements when faced with such a case (the comparison function returns `NaN`).

Reply:
(Typically something that would have been caught immediately with typescript)